hazelcast/src/main/java/com/hazelcast/client/impl/ClientEngineImpl.java

```diff
@@ -174,6 +179,21 @@ public void handleClientMessage(ClientMessage clientMessage, Connection connecti
         }
     }
 
+    private MessageTask resolveClientMessage(ClientMessage clientMessage, Connection connection) {
```

Comment:
I'm not that crazy about this part. It means that processing a client request/message needs to pass through additional logic and we have no idea how many task factories there are.